### PR TITLE
feat(useScroll): new Function

### PR DESCRIPTION
**Pull request by webfansplz**
#736 

@antfu 

It's doesn't support the `window` scroll, because I think it's too much for it (Maybe it would be better to support it in `useWindowScroll` ) 

Do you think need to do compatibility for `window` ? 

If don't need, Maybe it's better to rename it `useElementScroll`？

### CODE REVIEW

**Comment by antfu** [comment]
The function looks cool, thanks!

What's the problem of supporting window?

**Comment by webfansplz** [comment]
The main concern is browser compatibility, But i added simple support for window,make sure it works in the standard browsers.  (I tested Chrome, Firefox, and Safari. They work well) 😊

```ts
const eventTarget = (e.target === document ? (e.target as Document).documentElement : e.target) as HTMLElement
```

File: packages/core/useScroll/index.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+   *
+   */
+  onStop?: () => void
+
+  /**
+   * Enabled direction for scroll.
+   *
+   * @default ['x','y']
+   */
+  enabledDirection?: Array<'x'|'y'>
```
Comment (antfu) [inline packages/core/useScroll/index.ts:33]: When will this be used? Or I think we can just do it for both without the config, there should be much perf difference

File: packages/core/useScroll/index.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ * Reactive scroll.
+ *
+ * @see https://vueuse.org/useScroll
+ * @param {MaybeElementRef} element
+ * @param {() => void} [handler=noop]
+ * @param {UseScrollOptions} [options={}]
+ */
+export function useScroll(
+  element: MaybeElementRef,
+  handler: () => void = noop,
```
Comment (antfu) [inline packages/core/useScroll/index.ts:53]: Let's have this inside the option as `onScroll`

File: packages/core/useScroll/index.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+   */
+  throttle?: number
+
+  /**
+   * The check time when scrolling ends.
+   * This configuration will be setting to (throttle + checkStopTime) when the `throttle` is configured.
+   *
+   * @default 200
+   */
+  checkStopTime?: number
```
Comment (antfu) [inline packages/core/useScroll/index.ts:20]:
```suggestion
  idle?: number
```

File: packages/core/useScroll/index.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
+category: Sensors
+---
+
+# useScroll
+
+Reactive scroll
```
Comment (antfu) [inline packages/core/useScroll/index.md:7]:
```suggestion
Reactive scroll position and state
```

File: packages/core/useScroll/index.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+</template>
+
+<script>
+  setup() {
+    import { useScroll } from '@vueuse/core'
+    const scrollRef = ref<HTMLElement | null>(null)
+    const { x, y, scrolling, finished, arrivedStatus } = useScroll(scrollRef)
+    return { x, y, scrolling, finished, arrivedStatus  }
+  } 
+</script>
```
Comment (antfu) [inline packages/core/useScroll/index.md:24]:
```suggestion
<script setup>
import { useScroll } from '@vueuse/core'

const scrollRef = ref<HTMLElement | null>(null)

const { x, y, scrolling, finished, arrivedStatus } = useScroll(scrollRef)
</script>
```

And let's move this before `<template>`

File: packages/core/useScroll/index.md [hunk 6]
```diff
@@ -14,14 +14,11 @@ Reactive scroll
...
 
-<script>
-  setup() {
-    import { useScroll } from '@vueuse/core'
-    const scrollRef = ref<HTMLElement | null>(null)
-    const { x, y, scrolling, finished, arrivedStatus } = useScroll(scrollRef)
-    return { x, y, scrolling, finished, arrivedStatus  }
-  } 
+<script setup>
+	import { useScroll } from '@vueuse/core'
```
Comment (antfu) [inline packages/core/useScroll/index.md:18]: Let's use 2 spaces instead of tabs.

File: packages/core/useScroll/index.ts (hunk 1 above)
Comment (webfansplz) [reply packages/core/useScroll/index.ts:39]: By default, we do both. i think we should expose this config and let ppl setting it  to save perf overhead

Comment (antfu) [reply packages/core/useScroll/index.ts:39]: I don't see there is any overhead on that. It's just variable assignment, and we don't have extra event listener.

Comment (webfansplz) [reply packages/core/useScroll/index.ts:39]: Okay,Do you suggest that we remove this config and enable both by default ?

Comment (webfansplz) [reply packages/core/useScroll/index.ts]: removed~

**Pull request merged**